### PR TITLE
Add types

### PR DESCRIPTION
**Pull request by SalahAdDin**
This fixes the next issue: https://github.com/remarkjs/remark-slug/issues/18

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,15 +28,18 @@
     "Jayson Harshbarger <hypercubed@gmail.com>"
   ],
   "files": [
-    "index.js"
+    "index.js",
+    "types/index.d.ts"
   ],
+  "types": "types",
```
Comment (ChristianMurphy) [inline package.json:34]:
```suggestion
  "types": "types/index.d.ts",
```

File: package.json [hunk 2]
```diff
@@ -28,15 +28,18 @@
...
   "dependencies": {
     "github-slugger": "^1.0.0",
     "mdast-util-to-string": "^1.0.0",
     "unist-util-visit": "^2.0.0"
   },
   "devDependencies": {
     "browserify": "^17.0.0",
+    "dtslint": "^4.1.2",
```
Comment (ChristianMurphy) [inline package.json:42]:
```suggestion
    "dtslint": "^4.0.0",
```

File: package.json [hunk 3]
```diff
@@ -69,12 +73,13 @@
...
+      "types/index.d.ts"
     ]
   },
   "remarkConfig": {
     "plugins": [
       "preset-wooorm"
     ]
   }
-}
+}
```
Comment (ChristianMurphy) [inline package.json:85]:
```suggestion
}

```

File: types/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+      "module": "commonjs",
+      "lib": ["es2015"],
+      "strict": true,
+      "noEmit": true,
+      "baseUrl": ".",
+      "paths": {
+        "remark-slug": ["."]
+      }
+    }
+  }
```
Comment (ChristianMurphy) [inline types/tsconfig.json:12]: could you run prettier on this file?

```
prettier --write types/tsconfig.json
```

File: types/tslint.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+    "extends": "dtslint/dtslint.json",
+    "rules": {
+        "semicolon": false,
+        "whitespace": false
+    }
```
Comment (ChristianMurphy) [inline types/tslint.json:6]: could you run prettier on this file?

```
prettier --write types/tslint.json
```

**Review by ChristianMurphy — COMMENTED** [review]
Thanks @SalahAdDin!
Just a couple small updates and this should be good to go!

**Comment by SalahAdDin** [comment]
Done!

**Review by ChristianMurphy — APPROVED** [review]
Thanks @SalahAdDin!

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Thanks, released! https://github.com/remarkjs/remark-slug/releases/tag/6.1.0

**Pull request merged**